pkg/controller/common/random.go

```diff
@@ -23,3 +30,28 @@ func RandomBytes(length int) []byte {
 		true,  // allowRepeat
 	))
 }
+
+// RandomBytesRespectingLicense generates random bytes according to the Elastic license.
+// 1. For basic license, it generates a length of 24 random bytes using alphanumeric characters.
+// 2. For enterprise license, it allows control over both the length of the generated bytes and the character set.
+func RandomBytesRespectingLicense(ctx context.Context, client k8s.Client, namespace string, params operator.PasswordGeneratorParams) ([]byte, error) {
+	enabled, err := license.NewLicenseChecker(client, namespace).EnterpriseFeaturesEnabled(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if enabled {
+		data, err := password.Generate(
+			params.Length,
+			min(params.Length/4, len(params.Digits)),  // number of digits to include in the result
+			min(params.Length/4, len(params.Symbols)), // number of symbols to include in the result
+			false, // noUpper
+			true,  // allowRepeat
+		)
```

Comment:
This ignores the allowed characters parameter. Why do we introduce two parameters to control password generation if one of them does not actually work? I see two options either you adjust the operator flag to something the generator actually supports or you choose a different password generator implementation. But having a flag that implies that users can determine the allowed characters, when in fact they cannot is highly misleading.